File: Data Structures & Algorithms/binary-tree-right-side-view/submission-12.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */

class Solution {
public:
    vector<int> rightSideView(TreeNode* root) {
        if (!root) {
            return {};
        }

        std::queue<TreeNode*> q;
        q.push(root);
        vector<int> res;

        while (!q.empty()) {
            int qLen = q.size();
            int right;

            for (int i = 0; i < qLen; i++) {
                TreeNode* cur = q.front();
                q.pop();

                right = cur->val;
                if (cur->left != nullptr) q.push(cur->left);
                if (cur->right != nullptr) q.push(cur->right);
            }

            res.push_back(right);
        }

        return res;
    }
};

File: Data Structures & Algorithms/count-good-nodes-in-binary-tree/submission-12.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:
    def goodNodes(self, root: TreeNode) -> int:
        def dfs(node, maxVal):
            res = 0

            if not node:
                return 0

            if node.val >= maxVal:
                res += 1

            res += dfs(node.left, max(maxVal, node.val))
            res += dfs(node.right, max(maxVal, node.val))

            return res

        return dfs(root, -float("inf"))